src/transformers/modeling_tf_utils.py

```diff
@@ -533,9 +537,14 @@ def load_tf_weights(model, resolved_archive_file, _prefix=None):
                             # If the two shapes are not compatible we raise an issue
                             try:
                                 array = np.reshape(saved_weight_value, K.int_shape(symbolic_weight))
-                            except AssertionError as e:
-                                e.args += (K.int_shape(symbolic_weight), saved_weight_value.shape)
-                                raise e
```

Comment:
Can't that logic be re-used if `not ignore_mismatched_sizes` ?

Reply:
The error already contains the shape, so I didn't see the point of having it twice.